### PR TITLE
DBAL 2.13.1 and 3.1.0

### PR DESCRIPTION
**Pull request by beberlei**
Blogpost

### CODE REVIEW

File: source/blog/2021-04-19-dbal-2.13.1-3.1.0.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Last month [we released DBAL
+2.13.0](https://www.doctrine-project.org/2021/03/29/dbal-2.13.html) as an
+important push for the ecosystem towards DBAL 3 with an extensive deprecation
+layer and forwards compatibility.
+
+We made a few mistakes, given that the forward compatibility layer is quite complex.
+As such we have now released Doctrine DBAL 2.13.1 and 3.1.0 with two new APIs
+that improve the forward compatiblity.
+
+The problem lies in `Statement::execute`: 2.13.0 would return a `bool` and 3.0.0
```
Comment (morozov) [inline source/blog/2021-04-19-dbal-2.13.1-3.1.0.md:17]: A method referenced without quotes is quite weird in PHP. `Statement::execute()` should look better,

File: source/blog/2021-04-19-dbal-2.13.1-3.1.0.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+would return a `Result` from this method. This is a BC break that we cannot fix
```
Comment (morozov) [inline source/blog/2021-04-19-dbal-2.13.1-3.1.0.md:18]: It's not that we're trying to fix a BC break, we're trying to provide an upgrade path (via the FC layer).

File: source/blog/2021-04-19-dbal-2.13.1-3.1.0.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+with forward compatibility.
+
+As such we introduced two new APIs on `Statement` that replace `execute`.
```
Comment (morozov) [inline source/blog/2021-04-19-dbal-2.13.1-3.1.0.md:21]: `execute()` should look better.

File: source/blog/2021-04-19-dbal-2.13.1-3.1.0.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+```php
+$statement = $connection->prepare('SELECT * FROM tbl WHERE col = ?');
+$result = $statement->executeQuery();
+
+$rows = $result->fetchAllAssociative();
+```
+
+The DBAL 2.13 forward compatiblity layer supports both versions of all code and
+returns a Statement/Result hybrid that has all the APIs that the DBAL Statement
+had up until version 2.12. This way you can slowly move from the old to the new
```
Comment (morozov) [inline source/blog/2021-04-19-dbal-2.13.1-3.1.0.md:42]: I'd reword "slowly move" as "move at your own pace". Nobody _wants_ to move slow, some _have_ to move slower than the ecosystem.

File: source/blog/2021-04-19-dbal-2.13.1-3.1.0.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+API in your code.
+
+Thank you again to [mdumoulin](https://github.com/mdumoulin) for the work on
+improving the forward compatiblity and to [Sergei](https://twitter.com/srgmrzv)
```
Comment (morozov) [inline source/blog/2021-04-19-dbal-2.13.1-3.1.0.md:46]: You can use my full name. I think I'm not that famous as Rasmus or Fabien to be referenced by first name :)

**Review by morozov — COMMENTED** [review]
Looks great! Just a few nitpicks.

File: source/blog/2021-04-19-dbal-2.13.1-3.1.0.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+title: "New Release: Doctrine DBAL 2.13.1 and 3.1.0 with important Forward Compatibility fix"
+authorName: Benjamin Eberlei
+authorEmail: kontakt@beberlei.de
+permalink: /2021/04/19/dbal-2.13.1-3.1.0.html
+---
+
+Last month [we released DBAL
+2.13.0](https://www.doctrine-project.org/2021/03/29/dbal-2.13.html) as an
+important push for the ecosystem towards DBAL 3 with an extensive deprecation
+layer and forwards compatibility.
```
Comment (greg0ire) [inline source/blog/2021-04-19-dbal-2.13.1-3.1.0.md:11]: Below, you used "forward", and I think you should pick that one ("forward"). Also, should it read "with an extensive deprecation and forward compatibility layer"?

File: source/blog/2021-04-19-dbal-2.13.1-3.1.0.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+use Monolog\Logger;
+use Monolog\Handler\StreamHandler;
+
+$log = new Logger('doctrine');
+$log->pushHandler(new StreamHandler('deprecations.log', Logger::INFO));
+
+Deprecation::enableWithPsrLogger($log);
+```
+
+Or alternatively using PHPs global error handler:
```
Comment (greg0ire) [inline source/blog/2021-04-19-dbal-2.13.1-3.1.0.md:64]:
```suggestion
Or alternatively using PHP's global error handler:
```

**Review by morozov — APPROVED** [review]
